src/error.rs

```diff
@@ -0,0 +1,125 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at https://mozilla.org/MPL/2.0/.
+
+use std::fmt;
+use std::result;
+
+use failure::{Backtrace, Context, Fail};
+use rocket::http::{Status};
+use rocket::response::{Responder, Response};
+use rocket::{self, response, Request};
+use rocket_contrib::Json;
+
+use logging::MozlogLogger;
+
+pub type HandlerResult<T> = result::Result<T, HandlerError>;
+
+#[derive(Debug)]
+pub struct HandlerError {
+    inner: Context<HandlerErrorKind>,
+}
+
+#[derive(Clone, Eq, PartialEq, Debug, Fail)]
+pub enum HandlerErrorKind {
+    /// 400 Bad Request
+    #[fail(display = "Bad Request")]
+    BadRequest,
+
+    /// 404 Not Found
+    #[fail(display = "Not Found")]
+    NotFound,
+
+    /// 405 Method Not Allowed
+    #[fail(display = "Method Not Allowed")]
+    MethodNotAllowed,
+
+    /// 422 Unprocessable Entity
+    #[fail(display = "Unprocessable Entity")]
+    UnprocessableEntity,
+
+    /// 429 Too Many Requests
+    #[fail(display = "Too Many Requests")]
+    TooManyRequests,
+
+    /// 500 Internal Server Error
+    #[fail(display = "Internal Server Error")]
+    InternalServerError,
```

Comment:
Is there a way to provide detailed logging in DEV mode for `InternalServerError`s?

Reply:
Yes, that's what I'm going for :)